backend/src/main/java/atlantbh/restaurants/services/RestaurantService.java

```diff
@@ -0,0 +1,59 @@
+package atlantbh.restaurants.services;
+
+import atlantbh.restaurants.models.Restaurant;
+import atlantbh.restaurants.models.Review;
+import atlantbh.restaurants.models.filters.RestaurantFilterBuilder;
+import atlantbh.restaurants.models.sortkeys.RestaurantSortKeys;
+import atlantbh.restaurants.repositories.RestaurantRepository;
+import org.hibernate.Criteria;
+import org.hibernate.criterion.Restrictions;
+import org.hibernate.service.spi.ServiceException;
+import org.springframework.stereotype.Service;
+
+import java.beans.Transient;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.Collection;
+
+@Service
+public class RestaurantService extends BaseService<Restaurant, RestaurantSortKeys, RestaurantFilterBuilder, RestaurantRepository> {
+
+    public Collection<Restaurant> getPopularRestaurants(Integer numOfRestaurants) throws ServiceException {
+
+        try {
+            Criteria criteria = repository.getBaseCriteria();
+            criteria.add(Restrictions.sqlRestriction("1=1 ORDER BY RANDOM()"));
+            criteria.setMaxResults(numOfRestaurants);
+            return criteria.list();
+        } catch (Exception e) {
+            throw new ServiceException("Couldn't retrive popular restaurants", e);
+        }
+
+    }
+
+    @Transient
+    public Integer getReviewSize(Long id) throws ServiceException{
```

Comment:
not sure about this method.. to discuss

Reply:
Removed these methods in restaurantService and implemented them in reviewRepository 8c841a20e489a31c66d177c8682339f9d4f19de0